### ISSUE TITLE
SAFETY: Guidelines for unstripped bash compilation

### ISSUE THREAD

**Issue by pexes123** [strang1ato/nhi#28]
TLDR; Dockerfile or similar to reproduce ideally the same [executable](https://www.virustotal.com/gui/file/71c9c05805dc22ddecb63dbc426166fcf6df9b4514387c156086ee465cd1ac7d/details) as that on [PPA](https://launchpad.net/~strang1ato/+archive/ubuntu/default-bash-and-zsh/+build/23762501). 

In search for building hints I compared [PPA bash](https://launchpad.net/~strang1ato/+archive/ubuntu/default-bash-and-zsh/+sourcefiles/bash/99-5.1.16~ubuntu22.04/bash_99-5.1.16~ubuntu22.04.tar.gz) with [GNU 5.1.16](https://ftp.gnu.org/gnu/bash/bash-5.1.16.tar.gz):

`meld` reported differences in language files like `/po/*.gmo` and a couple of added tests and docs. Apart from these, there is `/debian` dir, with your rule overrides:
```
#!/usr/bin/make -f
%:
	dh $@

build:
	./configure --prefix=/usr --includedir=\${prefix}/include --mandir=\${prefix}/share/man --infodir=\${prefix}/share/info --sysconfdir=/etc --localstatedir=/var --libdir=\${prefix}/lib/x86_64-linux-gnu --libexecdir=\${prefix}/lib/x86_64-linux-gnu
	dh_auto_build 

override_dh_auto_configure:
	./configure --prefix=/usr --includedir=\${prefix}/include --mandir=\${prefix}/share/man --infodir=\${prefix}/share/info --sysconfdir=/etc --localstatedir=/var --libdir=\${prefix}/lib/x86_64-linux-gnu --libexecdir=\${prefix}/lib/x86_64-linux-gnu

override_dh_auto_test:
	echo
```

Maybe I am missing something, or I have compared against a wrong source ([google does not know any better](https://www.google.com/search?q=site:launchpad.net+bash+5.1.16)), but can not see any clues backing your unstripped claims ;) 
Het best thing I can do here is stop my search and ask for your guidance :)

FOOTNOTE: Bash processes many sensitive inputs, please share (and explain where considerable) the changes against the bash sources you had used.

**Comment by strang1ato** [comment]
It's not the issue with bash or zsh source code used on ubuntu, but rather ubuntu's compiled version of bash is stripped from data of dynamic symbol table (bss data is missing).

Tbh, I don't remember what flags I changed for builidng of my version of bash and zsh, since I was doing that couple of years ago, but my only objective was to compile bash and zsh with `.bss` data (as it is done on every other distro as I am aware of).

I don't plan on maintaining that ppa for newer version of ubuntu though, it's an overkill. if I want to keep nhi working on ubuntu I have to figure out how to get to bash `environ` without reading it's offset from bss table. If you want to try nhi on ubuntu, you can compile bash from source and use it instead.
